src/Microsoft.Diagnostics.ExtensionCommands/FindPointersInCommand.cs

```diff
@@ -0,0 +1,517 @@
+﻿using Microsoft.Diagnostics.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using static Microsoft.Diagnostics.ExtensionCommands.MAddressCommand;
+using Microsoft.Diagnostics.DebugServices;
+using System.IO;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    [Command(Name = "findpointersin", Help="Finds pointers to the GC heap within the given memory regions.")]
+    public class FindPointersInCommand : ExtensionCommandBase
+    {
+        private const string VtableConst = "vtable for ";
+
+        public IModuleService ModuleService { get; set; }
+        
+        public IMemoryRegionService MemoryRegionService { get; set; }
+        public ClrRuntime Runtime { get; set; }
+
+        [Option(Name = "--showAllObjects", Aliases = new string[] { "-a", "--all" }, Help = "Show all objects instead of only objects Pinned on the heap.")]
+        public bool ShowAllObjects { get; set; }
+
+        [Argument(Help = "The types of memory to search for pointers to the GC heap.")]
+        public string[] Regions { get; set; }
+
+        private int Width
+        {
+            get
+            {
+                int width = Console.WindowWidth;
+                if (width == 0)
+                    width = 120;
+                if (width > 256)
+                    width = 256;
+
+                return width;
+            }
+        }
+
+        public override void ExtensionInvoke()
+        {
+            if (Runtime is null)
+            {
+                WriteLine("ClrMD interfaces must be available for this command.");
+                return;
+            }
+
+            if (ModuleService is null)
+            {
+                WriteLine("IModuleService must be available for this command.");
+                return;
+            }
+
+            if (MemoryRegionService is null)
+            {
+                WriteLine("IMemoryRegionService must be available for this command.");
+                return;
+            }
+
+            if (Regions is null || Regions.Length == 0)
+            {
+                Console.WriteLine("Must specify at least one memory region type to search for.");
+                return;
+            }
+
+            Stopwatch sw = Stopwatch.StartNew();
+            PrintPointers(!ShowAllObjects, Regions);
+            WriteLine(sw.Elapsed.ToString());
+        }
+
+        private void PrintPointers(bool pinnedOnly, params string[] memTypes)
+        {
+            MAddressCommand address = new()
```

Comment:
Creating another command instance (MAddressCommand) explicitly and reference it from this command isn't recommended.  It would be better to share a helper class between them.

Reply:
No problem.  ClrMD based extension commands are meant to be composed (which is why I built it this way), but I'll factor it out.